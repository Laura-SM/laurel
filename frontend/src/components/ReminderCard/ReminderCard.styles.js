import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    width: 350,
    height: 150,
    borderWidth: 2,
    borderColor: '#93B1A7',
    borderRadius: 10,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginVertical: 10,
    marginLeft: 20,
    flexDirection: 'row',
  },
  image: {
    width: 100,
    height: 100,
    borderRadius: 10,
    backgroundColor: '#E5E5E5',
  },
  featuresContainer: {
    alignItems: 'flex-start',
    width: 150,
  },
  iconsContainer: {
    width: 170,
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-around',
    marginVertical: 10,
  },
});

export default styles;
