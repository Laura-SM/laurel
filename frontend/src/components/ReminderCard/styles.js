import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    width: 350,
    height: 150,
    borderWidth: 2,
    borderColor: '#93B1A7',
    borderRadius: 10,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginVertical: 10,
    marginLeft: 20,
    flexDirection: 'row',
  },
  image: {
    width: 120,
    height: 120,
    borderRadius: 10,
    backgroundColor: '#E5E5E5',
  },
  featuresContainer: {
    alignItems: 'flex-start',
    width: 180,
  },
  namesContainer: {
    width: 150,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    marginVertical: 10,
  },
  iconsContainer: {
    width: 180,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    marginVertical: 10,
  },
  iconContainer: {
    margin: 10,
  },
  boldText: {
    fontSize: 18,
    marginHorizontal: 5,
    fontWeight: 'bold',
  },
  text: {
    fontSize: 18,
    marginHorizontal: 5,
  },
});

export default styles;
