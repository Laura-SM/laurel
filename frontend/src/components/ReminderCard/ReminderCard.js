import React from 'react';
import {View, Text, Image} from 'react-native';
import globalStyles from '../../styles/global.styles';
import styles from './ReminderCard.styles';

const ReminderCard = ({plant}) => {
  const currentDate = new Date();
  const currentDateNoTime = new Date(
    Date.UTC(
      currentDate.getFullYear(),
      currentDate.getMonth(),
      currentDate.getDate(),
    ),
  );

  const waterDate = new Date(plant.nextWaterDate).toLocaleDateString();
  const mistDate = new Date(plant.nextMistDate).toLocaleDateString();
  const transplantDate = new Date(
    plant.nextTransplantDate,
  ).toLocaleDateString();

  return (
    <View style={styles.cardContainer}>
      <View style={styles.image} />
      <View style={styles.featuresContainer}>
        <View style={styles.iconsContainer}>
          <Text style={globalStyles.boldText}>{plant.name}</Text>
          <Text style={globalStyles.text}>{plant.room}</Text>
        </View>
        <View style={styles.iconsContainer}>
          {waterDate <= currentDateNoTime.toLocaleDateString() && (
            <Image source={require('../../icons/water24.png')} />
          )}
          {mistDate <= currentDateNoTime.toLocaleDateString() && (
            <Image source={require('../../icons/mist24.png')} />
          )}
          {transplantDate <= currentDateNoTime.toLocaleDateString() && (
            <Image source={require('../../icons/settings24.png')} />
          )}
        </View>
      </View>
    </View>
  );
};

export default ReminderCard;
