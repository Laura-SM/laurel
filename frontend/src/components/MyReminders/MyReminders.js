/* eslint-disable react-hooks/exhaustive-deps */
import React, {useEffect} from 'react';
import {View, Text, FlatList} from 'react-native';
import {connect} from 'react-redux';
import {loadPlants} from '../../redux/actions/plantsActionCreators';
import ReminderCard from '../ReminderCard/ReminderCard';
import globalStyles from '../../styles/global.styles';

const MyReminders = ({plants, userAccess, dispatch}) => {
  const currentDate = new Date();
  const currentDateNoTime = new Date(
    Date.UTC(
      currentDate.getFullYear(),
      currentDate.getMonth(),
      currentDate.getDate(),
    ),
  );
  const renderReminderCard = ({item}) => <ReminderCard plant={item} />;

  const reminderPlants = plants.filter(plant => {
    return (
      userAccess.user.plants.includes(plant._id) &&
      (new Date(plant.nextWaterDate).getTime() <= currentDateNoTime.getTime() ||
        new Date(plant.nextMistDate).getTime() <= currentDateNoTime.getTime() ||
        new Date(plant.nextTransplantDate).getTime() <=
          currentDateNoTime.getTime())
    );
  });

  useEffect(() => {
    if (plants.length) {
      dispatch(loadPlants());
    }
  }, [plants]);

  console.log(reminderPlants);

  return (
    <FlatList
      style={globalStyles.mainContainer}
      ListEmptyComponent={
        <Text style={globalStyles.text}>You don't have any reminder!</Text>
      }
      ListHeaderComponent={
        <View style={globalStyles.headerContainer}>
          <Text style={globalStyles.titleText}>My reminders</Text>
        </View>
      }
      data={reminderPlants}
      renderItem={renderReminderCard}
      keyExtractor={plant => plant._id}
    />
  );
};

function mapStateToProps({plants, userAccess}) {
  return {
    plants,
    userAccess,
  };
}

export default connect(mapStateToProps)(MyReminders);
