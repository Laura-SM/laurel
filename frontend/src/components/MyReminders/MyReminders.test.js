import React from 'react';
import MyReminders from './index';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import ReminderCard from '../ReminderCard/index';

describe('Given a MyReminders component', () => {
  const mockStore = configureStore([thunk]);
  let store = mockStore({
    plants: [
      {name: 'kentia', _id: '1a'},
      {name: 'potus', _id: '2a'},
    ],
    userAccess: {
      user: {
        plants: ['1a'],
      },
    },
  });

  describe('When is invoked', () => {
    test('Then it is renders correctly', () => {
      const wrapper = render(
        <Provider store={store}>
          <MyReminders>
            <ReminderCard />
          </MyReminders>
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When MyReminders function is invoked with an empty plants list', () => {
    test('Then is rendered whitout plants', () => {
      store = mockStore({
        plants: [],
        userAccess: {
          user: {
            plants: [],
          },
        },
      });
      const wrapper = render(
        <Provider store={store}>
          <MyReminders>
            <ReminderCard />
          </MyReminders>
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
