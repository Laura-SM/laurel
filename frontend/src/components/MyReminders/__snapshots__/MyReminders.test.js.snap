// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a MyReminders component When MyReminders function is invoked with an empty plants list Then is rendered whitout plants 1`] = `
<RCTScrollView
  ListEmptyComponent={
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginHorizontal": 20,
        }
      }
    >
      You don't have any reminder!
    </Text>
  }
  ListHeaderComponent={
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "margin": 20,
          }
        }
      >
        My reminders
      </Text>
    </View>
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "margin": 20,
            }
          }
        >
          My reminders
        </Text>
      </View>
    </View>
    <Text
      onLayout={[Function]}
      style={
        Object {
          "fontSize": 18,
          "marginHorizontal": 20,
        }
      }
    >
      You don't have any reminder!
    </Text>
  </View>
</RCTScrollView>
`;

exports[`Given a MyReminders component When is invoked Then it is renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginHorizontal": 20,
        }
      }
    >
      You don't have any reminder!
    </Text>
  }
  ListHeaderComponent={
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "margin": 20,
          }
        }
      >
        My reminders
      </Text>
    </View>
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "margin": 20,
            }
          }
        >
          My reminders
        </Text>
      </View>
    </View>
    <Text
      onLayout={[Function]}
      style={
        Object {
          "fontSize": 18,
          "marginHorizontal": 20,
        }
      }
    >
      You don't have any reminder!
    </Text>
  </View>
</RCTScrollView>
`;
